.github/workflows/osx.yml

```
@@ -58,7 +61,7 @@ jobs:
         uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
         with:
           repository: kokkos/kokkos
-          ref: 4.3.01
+          ref: $KOKKOS_VERSION
```

osx-ci doesn't like this variable, maybe braces are needed?
```suggestion
          ref: ${KOKKOS_VERSION}
```

---

Yep that looks reasonable!

---

I'll do a separate commit as I am not sure how to sign-off otherwise...

---

I think since the checkout is performed by an action this made including the context necessary (if it was handled on the runner then the Linux shell syntax would've been fine), [using-the-env-context-to-access-environment-variable-values](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#using-the-env-context-to-access-environment-variable-values)